.github/workflows/ci.yml

```diff
@@ -3,7 +3,7 @@ on:
   push:
     branches: [main]
   pull_request:
-    branches: [main, 0.3]
+  merge_group:
```

Comment:
Should this not be nested?

Also does this mean CI is _**required**_? Because I think a few of the jobs don't work if not targeting main

Reply:
> Should this not be nested?

No, `merge_group` is completely separated from `pull_request`

> Also does this mean CI is _**required**_? Because I think a few of the jobs don't work if not targeting main

No. the `required` status will only be added when targeting a protected branch. If you, however, targeting a protected branch without running the CI you're not able to merge that PR at all.